### PR TITLE
Implement getSharedLibs() using "debug/elf"

### PR DESCRIPTION
**Pull request by mkhon**
There is also a support for NANOS_TARGET_ROOT env variable (if set - points to the Linux chroot environment) - it is currently commented out.

There's corresponding code in mkfs (also ifdef'ed out under DEV) which also implements NANOS_TARGET_ROOT.

If NANOS_TARGET_ROOT is compiled-in ops/mkfs can be used on macOS to build images for dynamically linked Linux programs (given that chroot environment is provided).

chroot environment can be pre-created on Linux with e.g. "debootstrap"

### CODE REVIEW

**Comment by eyberg** [comment]
does that env var force a chroot on osx or just optional?

**Comment by mkhon** [comment]
@eyberg it is optional (if not set ops/mkfs behave as before) and it is even not compiled-in by default (although I would prefer to have it compiled-in by default)

File: lepton/elf.go [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+			break;
+		}
+	}
+
+	// fmt.Printf("%s -> %s\n", path, targetPath)
+	return targetPath, nil
+}
+
+func findLib(libDirs []string, path string) (string, error) {
+	if path[0] == '/' {
```
Comment (tijoytom) [inline lepton/elf.go]: When is path[0] == '/'? Is it when on Mac?

Comment (mkhon) [reply lepton/elf.go]: @tijoytom I've just tried to follow ld.so(8) man page:

```
If a shared object dependency does not contain a slash, then it is searched for in the following order:
...
```

**Pull request merged**